packages-exp/auth-exp/src/core/user/reload.ts

```diff
@@ -30,23 +31,32 @@ import { getModularInstance } from '@firebase/util';
 
 export async function _reloadWithoutSaving(user: UserInternal): Promise<void> {
   const auth = user.auth;
-  const idToken = await user.getIdToken();
-  const response = await _logoutIfInvalidated(
-    user,
-    getAccountInfo(auth, { idToken })
-  );
+  let coreAccount: APIUserInfo = {};
+
+  if (user.stsTokenManager.isPassthroughMode) {
+    // The uid === localId === id token's `sub` claim
+    coreAccount.localId = user.uid;
+  } else {
+    const idToken = await user.getIdToken();
+    const response = await _logoutIfInvalidated(
+      user,
+      getAccountInfo(auth, { idToken })
+    );
 
-  _assert(response?.users.length, auth, AuthErrorCode.INTERNAL_ERROR);
+    _assert(response?.users.length, auth, AuthErrorCode.INTERNAL_ERROR);
 
-  const coreAccount = response.users[0];
+    coreAccount = response.users[0];
+  }
 
   user._notifyReloadListener(coreAccount);
 
   const newProviderData = coreAccount.providerUserInfo?.length
     ? extractProviderData(coreAccount.providerUserInfo)
     : [];
 
-  const providerData = mergeProviderData(user.providerData, newProviderData);
+  const providerData = !user.stsTokenManager.isPassthroughMode
```

Comment:
nit: i would remove the `!` and just switch the order of the if/else:

```
const providerData = user.stsTokenManager.isPassthroughMode ? [] : mergeProviderData(user.providerData, newProviderData);
```
a little easier to reason about

Reply:
Fair, done!